docs/glossary.md

```diff
@@ -0,0 +1,258 @@
+the bundle.
+
+**Data Item** - An individual piece of data within a bundle. Each data item has
+its own ID, signature, tags, and data, similar to a transaction but more
+lightweight.
+
+**Data Item ID** - A unique identifier for a data item within a bundle,
+calculated from the data item's signature.
+
+**Nested Bundle** - A bundle contained within another bundle. Identified by
+having a parent_id field pointing to the containing bundle.
+
+**Parent ID** - The identifier of the containing bundle or transaction for a
+data item. For data items in nested bundles, points to the immediate parent
+bundle. For top-level data items, equals the root transaction ID.
+
+**Root Transaction ID** - The top-level Arweave transaction ID containing a
+bundle or data item. Used to retrieve the raw data from Arweave.
+
+**Unbundling** - The process of extracting and indexing individual data items
+from a bundle transaction.
+
+## AR.IO Gateway Concepts
+
+**ArNS (Arweave Name System)** - A decentralized naming system that maps
+human-readable names to Arweave transaction IDs, similar to DNS for the
+permaweb.
+
+**Contiguous Data** - Complete, reassembled data from all chunks of a
+transaction. The gateway stores metadata about contiguous data availability and
+verification status.
+
+**Data Verification** - The process of cryptographically verifying that
+retrieved data matches its data root hash, ensuring data integrity.
+
+**Observer** - A gateway component that monitors and reports on the health and
+behavior of other gateways in the AR.IO network.
+
+**X-AR-IO Headers** - HTTP headers added by the gateway providing metadata about
+data verification, caching, and trust status.
+
+**X-ArNS Headers** - HTTP response headers added during ArNS name resolution
+that provide metadata about the resolution process, including the resolved
+transaction ID, TTL, process ID, and undername information.
+
+## Databases
+
+**Bundles Database** - SQLite database storing ANS-104 bundle metadata, data
+items, and their relationships. Includes retry tracking for bundle processing.
+
+**Core Database** - Primary SQLite database containing blocks, transactions,
+transaction tags, stable/new data indexes, and the migrations table that tracks
+applied database schema changes.
+
+**Data Database** - SQLite database tracking contiguous data availability,
+verification status, and retry attempts.
+
+**Moderation Database** - SQLite database managing content blocking and
+filtering rules.
+
+## Data Storage Architecture
+
+**Cache Store** - High-speed storage layer (Redis or filesystem) for frequently
+accessed data chunks and headers.
+
+**Chunk Data Store** - Backend storage for transaction chunks. Supports multiple
+implementations including filesystem and S3.
+
+**Contiguous Data Store** - Storage backend for complete transaction data.
+Manages both data files and verification metadata.
+
+**Verification Priority** - Numeric value determining the order of data
```

Comment:
These verification entries might benefit from their own section. They feel a bit awkwardly positioned among the others in in this section.